tripleblindmarket/private-kit
# Introduce gps and bte ios targets

Why:
We would like to introduce bluetooth to the application for bluetooth
based contact tracing. Given the privacy concerns of having both
location and bluetooth data available to the same application, google
and apple do not allow for both location and bluetooth to be available
at the same time.

Hence in order to service both location and bluetooth
we will need to provide separate apps with separate info.plists and
bundle identifiers. At the same time we would like to share a much
application logic between the two applications as to reduce the
maintenance and organizational overhead. Ideally as close to 100% of the
javascript layer is common and for the ios layer we only need to change
the info.plists.

This commit:

Introduces two build targets, GPS and BTE, to the xcode project which
are setup to read from the corresponding groups for config and info.plist
A COVIDSafePathsConfig file was introduced to expose a
'getTracingStrategy' function to the react native side so that the js
runtime can determine which tracing strategy the bundle will be using.
This will allow us flexibility to programmatically determine which
tracing strategy is being used either within a given react component or
to conditional render components altogether. This also allows use to
maximize the amount of shared native code.

Three schemes where added for each target, a Development, a Staging, and
a Production.

To build the corresponding targets via the react native cli, you can
pass the --scheme flag with the corresponding scheme, e.g:

```
react-native run-ios --scheme "GPS_Development"
react-native run-ios --scheme "BTE_Development"
```

We updated the `yarn run-ios` script in package.json to two scripts for
for the gps based build and the bte based build, `yarn run-ios-bte` and
`yarn run-ios-gps` respectively

Note that theres a fair amount of location information storing logic
currently in AppDelegate, a future commit might want to refactor this so
that the gps specific code is only built in the GPS target builds.

### GPS
<img width="584" alt="Screen Shot 2020-05-20 at 4 19 14 PM" src="https://user-images.githubusercontent.com/16049495/82497012-9950e980-9abb-11ea-87b8-07bdf10e88d1.png">


### BTE
<img width="584" alt="Screen Shot 2020-05-20 at 4 17 47 PM" src="https://user-images.githubusercontent.com/16049495/82497021-9e159d80-9abb-11ea-89d2-70ebcc6715dd.png"> 


### Testing

Verify you can run both versions of the app.
- reinstall your pods, clean build folder, remove derived data, etc.
- run `yarn run-ios-gps`
- run `yarn run-ios-bte`
